### DISCUSSION TITLE
Update the message shown to the user, when enrollment date is over - high priority

### DISCUSSION THREAD

**Discussion by mmanimegalai** [Sunbird-Ed/Community#41] [Enhancement request]
This is a DIKSHA request **with high priority**

**Context:**

Currently when a user tries to access a course (on the portal & mobile app) that has a batch with enrollment date crossed, different types of error messages are shown which is misleading. 
1. Scenario:1 when a user lands on the toc page, s/he sees a message “You cannot enrol into this batch, as it is closed” on portal and "Enrollments for this batch closed on <date format 1> on mobile app
2. Scenario:2 When a user clicks any of the course modules on the toc page, s/he sees a message “Enrolments for this batch closed on <Date>”  on portal and mobile with different date formats
3. Scenario:3 When a user expands the batch section on the right side on portal, s/he sees a message “You cannot enrol into this batch, as it is closed”

**Required enhancement:**
In all the above scenarios, 
1. When the enrollment date is over, an appropriate error message should be shown that clearly says that the enrollment date is crossed. 
2. Also this should be consistent across portal and mobile app

**Comment by pramodkvarma** [comment]
Aren't all the messages in a bundle outside software code? If so, this should be easy to configure, right? @sharathkashyap @SanthoshVasabhaktula

**Reply by sharathkashyap** [reply]
@pramodkvarma We are externalising resource Bundle in current Sprint (4.10.3). This is a P1 enhancement to be implemented.

**Reply by pramodkvarma** [reply]
Ok great! If someone wants to do a PR for earlier release, please review and get that in.

**Comment by sharathkashyap** [comment]
@mmanimegalai The original requirement was a case of mis handling an error use case. We have corrected the implementation as this use case was mapped wrongly to a different error message. We had to make a code change to correct error handling. We have fixed the issue in 4.10.2

cc: @luckynvdu

**Reply by luckynvdu** [reply]
@mmanimegalai This has been taken as part of this ticket https://project-sunbird.atlassian.net/browse/SB-30483 in 4.10.2

@sharathkashyap
